dss-apps/dss-datamodel-center/dss-datamodel-indicator-management/src/main/java/com/webank/wedatasphere/dss/datamodel/indicator/service/impl/IndicatorServiceImpl.java

```diff
@@ -0,0 +1,539 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Resource;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+@Service
+public class IndicatorServiceImpl extends ServiceImpl<DssDatamodelIndicatorMapper, DssDatamodelIndicator> implements IndicatorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IndicatorServiceImpl.class);
+
+
+    private final Gson gson = new Gson();
+
+    @Resource
+    private ModelMapper modelMapper;
+
+    @Resource
+    private IndicatorContentService indicatorContentService;
+
+    @Resource
+    private IndicatorVersionService indicatorVersionService;
+
+    @Resource
+    private IndicatorQueryMapper indicatorQueryMapper;
+
+
+    @Resource
+    private LinkisDataAssetsRemoteClient linkisDataAssetsRemoteClient;
+
+    @Resource
+    private AssertsSyncService assertsSyncService;
+
+
+    @Resource
+    private DatamodelReferencService datamodelReferencService;
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public Long addIndicator(IndicatorAddVO vo, String version) throws ErrorException {
+        if (getBaseMapper().selectCount(Wrappers.<DssDatamodelIndicator>lambdaQuery().eq(DssDatamodelIndicator::getName, vo.getName())
+                .or().eq(DssDatamodelIndicator::getFieldIdentifier, vo.getFieldIdentifier())) > 0) {
+            LOGGER.error("errorCode : {}, indicator name or field identifier can not repeat, name : {}", ErrorCode.INDICATOR_ADD_ERROR.getCode(), vo.getName());
+            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_ADD_ERROR.getCode(), "indicator name or field identifier can not repeat");
+        }
+        //校验引用情况
+        if (datamodelReferencService.indicatorReferenceCount(vo.getName())>0
+                ||datamodelReferencService.indicatorReferenceCount(vo.getFieldIdentifier())>0){
+            LOGGER.error("errorCode : {}, indicator name can not  referenced", ErrorCode.INDICATOR_ADD_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_ADD_ERROR.getCode(), "indicator name can not referenced");
+        }
+
+
+        DssDatamodelIndicator newOne = modelMapper.map(vo, DssDatamodelIndicator.class);
+        //基本信息中设置版本号
+        newOne.setVersion(version);
+        newOne.setCreateTime(new Date());
+        newOne.setUpdateTime(new Date());
+        if (!save(newOne)) {
+            LOGGER.error("errorCode : {}, add indicator error", ErrorCode.INDICATOR_ADD_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_ADD_ERROR.getCode(), "add indicator error");
+        }
+
+        indicatorContentService.addIndicatorContent(newOne.getId(), version, vo.getContent());
+
+        //同步atlas
+        CreateModelTypeResult result = assertsSyncService.syncCreateModel(
```

Comment:
publisher.publishEvent(  new CreateModelEvent(this
                , DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                , vo.getFieldIdentifier()
                , ClassificationConstant.INDICATOR));